chia/wallet/vc_wallet/cr_cat_wallet.py

```diff
@@ -871,28 +873,41 @@ async def match_hinted_coin(self, coin: Coin, hint: bytes32) -> bool:
         This matches coins that are either CRCATs with the hint as the inner puzzle, or CRCATs in the pending approval
         state that will come to us once claimed.
         """
-        return (
+        authorized_providers_hash: bytes32 = Program.to(self.info.authorized_providers).get_tree_hash()
+        proofs_checker_hash: bytes32 = self.info.proofs_checker.as_program().get_tree_hash()
+        hint_inner_hash: bytes32 = construct_cr_layer_hash(
+            authorized_providers_hash,
+            proofs_checker_hash,
+            hint,
+        )
+        if (
             construct_cat_puzzle(
-                CAT_MOD,
+                Program.to(CAT_MOD_HASH),
```

Comment:
The change further below changes the call to `.get_tree_hash_precalc` to escape this value from the hash calculation.  So instead of hashing `CAT_MOD` every time, we now tell the tree hash function that we've already done it and it can skip doing that hash when it sees it.